src/dotnet-uninstall/Shared/Exceptions/ExceptionHandler.cs

```diff
@@ -16,6 +19,17 @@ public static Action HandleException(Action action)
                 {
                     PrintExceptionMessage(e.Message);
                 }
+                catch (Win32Exception e)
```

Comment:
This catch is too far, you should catch it in the immediate caller method(or the adapter layer). And handle it there. Following [this](https://docs.microsoft.com/en-us/dotnet/standard/exceptions/best-practices-for-exceptions)

Currently it means the whole stack has the possibly to throw NATIVE_ERROR_CODE_CANCELED. It is hard to refactor. In higher level, you let the adapter error handling leaked to other place.

Reply:
I agree. Will move this to a more proper place.

Reply:
Done.